cw-rct201/src/Components/Admin/AdminSingleProduct.tsx

```diff
@@ -1,518 +0,0 @@
-            
-              {/* This Drawer is For the Price Updateion */}
-         
-              {/* this Drawer is for the image updation */}
-             
-            </AlertDialogBody>
-
-            <AlertDialogFooter>
-              <Button
-                colorScheme="red"
-                ref={cancelRef}
-                onClick={() => setUpdateOpen(false)}
-              >
-                Cancel
-              </Button>
-              {/* <Button colorScheme="red" onClick={() => setUpdateOpen(false)} ml={3}>
-                Delete
-              </Button> */}
-
-              {/* Here is the categoruyl updation concept */}
-               <Drawer
-               
-      isOpen={categoryDrawerOpen}
-      placement='top'
-      onClose={()=>setCategoryDrawerOpen(false)}
-
-    >
-      <DrawerOverlay />
-      <DrawerContent>
-        <DrawerCloseButton />
-        <DrawerHeader>  Current category :  {category} <br /> Enter New Category For the Product </DrawerHeader>
-
-        <DrawerBody>
-          <Input onChange={(e)=>setNewCat(e.target.value)} placeholder='Type here...' />
-        </DrawerBody>
-
-        <DrawerFooter>
-          <Button variant='outline' mr={3} onClick={()=>setCategoryDrawerOpen(false)}>
-            Cancel
-          </Button>
-          <Button onClick={()=>updateCategory(id)} colorScheme='blue'>Save</Button>
-        </DrawerFooter>
-      </DrawerContent>
-    </Drawer>
-            </AlertDialogFooter>
-          </AlertDialogContent>
-        </AlertDialogOverlay>
-      </AlertDialog>
-    <Box
-      p={"10px 0px 10px 0px"}
-      borderRadius={"1rem"}
-      boxShadow={"rgba(0, 0, 0, 0.35) 0px 5px 15px;"}
-      textAlign={"center"}
-    >
-      <Box
-        margin={"auto"}
-        width={"40%"}
-        backgroundColor={"white"}
-        backdropFilter={"blur(20px)"}
-        borderRadius={"2rem"}
-      >
-        <Avatar size={"xl"} src={image1} />
-      </Box>
-      <Text mt={1} fontWeight={"bold"} fontSize={"16px"}>
-        {" "}
-        Name : {name}
-      </Text>
-      <Text mt={1} fontWeight={"bold"} fontSize={"16px"}>
-        {" "}
-        Category : {category}
-      </Text>
-      <Text mt={1} fontWeight={"bold"} fontSize={"16px"}>
-        Product Id : {id}
-      </Text>
-      <Text mt={1} fontWeight={"bold"} fontSize={"16px"}>
-        Price : ₹ {cost}
-      </Text>
-      <Flex mt={3} margin={"auto"} w={"60%"} justifyContent={"space-evenly"}>
-        <Button colorScheme="blue" onClick={() => setUpdateOpen(true)}>
-          <EditIcon />
-        </Button>
-        <Button onClick={()=>setDeleteOpen(true)} colorScheme="blue">
-          <DeleteIcon />
-        </Button>
-      </Flex>
-      <>
-   
-{/* This Is The Delete Part oF the admin Product */}
-     
-
-      {/* This is wehre we give users an option to do update name,price,category */}
-       
-      </>
-    </Box>
-    </>
-  );
-};
-
-export default AdminSingleProduct;
```

Comment:
This file is too big. Try to break down this file into a few smaller components. The ideal size of a component is 100 lines. If component is becoming lengthier than that, then break it down into smaller components.

Reply:
Yes Sir I have Break the Fille into Smaller Components Now